src/apiClient/abstractClient.ts

```diff
@@ -0,0 +1,96 @@
+import { ethers } from "ethers";
+import { ChainId } from "../transfers-history/adapters/web3/model";
+import {
+  CoingeckoDataReturnType,
+  SuggestedFeeReturnType,
+  BridgeLimitsReturnType,
+  AcrossBridgeStatisticsType,
+} from "./types";
+
+/**
+ * AbstractApiClient is an abstract class that defines the interface for the API client.
+ */
+export default abstract class AbstractApiClient {
```

Comment:
im not sure the intent of this, but i assume its so that you can create a mock with the same interface as the production class. i thought maybe there was a way you could avoid this by just extending and overriding the production class itself in the mock, but id have to play around with it to be sure.  it just seems like theres a lot of redundant code. not a criticism just looking for a way to simplify!

Reply:
I totally hear this argument. There's a bit of stylistic preference here.

Simple interfaces can be helpful as a clean way to document the expected behavior of the class. However, they definitely add boilerplate, especially in cases where there's only one implementation of the interface used in production (like this).

Either way y'all decide to go, LGTM!